Maple2.Server.Game/Manager/ShopManager.cs

```diff
@@ -85,162 +85,211 @@ private bool TryGetShopData(int shopId, [NotNullWhen(true)] out CharacterShopDat
         return accountShopData.TryGetValue(shopId, out data) || characterShopData.TryGetValue(shopId, out data);
     }
 
+    private bool TryRemoveShopData(int shopId) {
+        if (accountShopData.Remove(shopId)) {
+            return true;
+        }
+        return characterShopData.Remove(shopId);
+    }
+
     private bool TryGetShopItemData(int shopId, [NotNullWhen(true)] out IDictionary<int, CharacterShopItemData>? data) {
         return accountShopItemData.TryGetValue(shopId, out data) || characterShopItemData.TryGetValue(shopId, out data);
     }
 
+    /// <summary>
+    /// Load shop.
+    /// </summary>
+    /// <param name="shopId">Shop ID</param>
+    /// <param name="npcId">Npc ID. Optional parameter. This only is used for displaying the NPC's name on the shop window top bar.</param>
     public void Load(int shopId, int npcId = 0) {
-        Shop? shop = session.FindShop(shopId);
-        if (shop == null) {
-            logger.Warning("Shop {Id} has not been implemented", shopId);
+        if (!session.ServerTableMetadata.ShopTable.Entries.TryGetValue(shopId, out ShopMetadata? metadata)) {
+            logger.Warning("Shop {Id} not found", shopId);
             return;
         }
 
-        if (shop.RestockData != null) {
-            shop = GetInstancedShop(shop);
+        if (metadata.EnableReset) {
+            activeShop = GetInstancedShop(metadata);
+        } else {
+            activeShop = new Shop(metadata) {
+                Items = GetShopItems(metadata),
+            };
         }
 
-        activeShop = shop;
         shopNpcId = npcId;
 
         session.Send(ShopPacket.Open(activeShop, shopNpcId));
-        session.Send(ShopPacket.LoadItems(shop.Items.Values));
-        if (!shop.DisableBuyback) {
+        session.Send(ShopPacket.LoadItems(activeShop.Items.Values));
+        if (!metadata.IsOnlySell) {
             session.Send(ShopPacket.BuyBackItemCount((short) buyBackItems.Count));
             if (buyBackItems.Count > 0) {
-                session.Send(ShopPacket.LoadBuyBackItem((BuyBackItem[]) buyBackItems.Values));
+                session.Send(ShopPacket.LoadBuyBackItem(buyBackItems.Values.ToArray()));
             }
         }
     }
 
-    private Shop GetInstancedShop(Shop serverShop) {
-        if (instancedShops.TryGetValue(serverShop.Id, out Shop? instancedShop)) {
+    private Shop GetInstancedShop(ShopMetadata metadata) {
```

Comment:
**suggestion:** Consider breaking down GetInstancedShop into smaller methods

This method has grown complex and handles multiple responsibilities. Consider refactoring it into smaller, more focused methods to improve readability and maintainability.